lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -1154,7 +1154,7 @@ private function getCommitOrder(array $entityChangeSet = null)
 
                 $joinColumns = reset($assoc['joinColumns']);
 
-                $calc->addDependency($targetClass->name, $class->name, (int)empty($joinColumns['nullable']));
+                $calc->addDependency($targetClass->name, $class->name, (int) ! ( ! isset($joinColumns['nullable']) || $joinColumns['nullable'] === true));
```

Comment:
Lots of negations. Also, this is only checking the first join column... Possibly requiring to add tests around this area. :-\